nano/node/ledger_walker.cpp

```diff
@@ -0,0 +1,70 @@
+#include <nano/lib/blocks.hpp>
+#include <nano/lib/errors.hpp>
+#include <nano/node/ledger_walker.hpp>
+#include <nano/secure/blockstore.hpp>
+#include <nano/secure/ledger.hpp>
+
+#include <utility>
+
+nano::ledger_walker::ledger_walker (nano::ledger const & ledger_a) :
+	ledger{ ledger_a },
+	walked_blocks{},
+	blocks_to_walk{}
+{
+	if (ledger.store.init_error ())
+	{
+		throw nano::error{ "Ledger store initialization error" };
```

Comment:
This should be a debug_assert since it's a correctness issue the caller can check. We don't use exceptions so if this had needed to signal an error it should either return it or use a by-reference error code.